phoenix-queryserver/src/test/java/org/apache/phoenix/queryserver/server/PhoenixRemoteUserExtractorTest.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.phoenix.queryserver.server;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.apache.calcite.avatica.server.RemoteUserExtractionException;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.apache.hadoop.security.authorize.AuthorizationException;
+import org.apache.hadoop.security.authorize.ProxyUsers;
+import org.apache.phoenix.queryserver.server.QueryServer.PhoenixRemoteUserExtractor;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * Tests for the RemoteUserExtractor Method Avatica provides for Phoenix to implement.
+ */
+public class PhoenixRemoteUserExtractorTest {
+  private static final Logger LOG = LoggerFactory.getLogger(PhoenixRemoteUserExtractorTest.class);
+
+  @Test
+  public void testUseDoAsSuccess() {
+    HttpServletRequest request = mock(HttpServletRequest.class);
+    when(request.getRemoteUser()).thenReturn("proxyserver");
+    when(request.getParameter("doAs")).thenReturn("enduser");
+    when(request.getRemoteAddr()).thenReturn("localhost:1234");
+
+    Configuration conf = new Configuration(false);
+    conf.set("hadoop.proxyuser.proxyserver.groups", "*");
+    conf.set("hadoop.proxyuser.proxyserver.hosts", "*");
+    conf.set("phoenix.queryserver.doAs.enabled", "true");
+    ProxyUsers.refreshSuperUserGroupsConfiguration(conf);
+
+    PhoenixRemoteUserExtractor extractor = new PhoenixRemoteUserExtractor(conf);
+    try {
+      assertEquals("enduser", extractor.extract(request));
+    } catch (RemoteUserExtractionException e) {
+      LOG.info(e.getMessage());
+    }
+  }
+
+  @Test
+  public void testDoNotUseDoAs() {
```

Comment:
To test this code if you take my above suggestion, you could make a new method in QueryServer which does

```java
Builder setRemoteUserExtractorIfNecessary(Builder b, Configuration conf) {
  if (conf.getBoolean(QueryServices.QUERY_SERVER_DOAS_ENABLED_ATTRIB,
          QueryServicesOptions.DEFAULT_QUERY_SERVER_DOAS_ENABLED)) {
    return builder.withRemoteUserExtractor(new PhoenixRemoteUserExtractor(getConf()));
  }
  return builder;
}
```

This would let you easily mock the Builder and verify that your extractor is configured when the property is set to "true".